ui/tickets_activity_page.go

```diff
@@ -0,0 +1,162 @@
+		{Text: "Revoked"},
+	}, 1)
+
+	return pg
+}
+
+func (pg *ticketsActivityPage) Layout(gtx layout.Context) layout.Dimensions {
+	c := pg.common
+	body := func(gtx C) D {
+		page := SubPage{
+			title: "Ticket activity",
+			back: func() {
+				c.changePage(PageTickets)
+			},
+			body: func(gtx C) D {
+				walletID := c.info.Wallets[pg.walletDropDown.SelectedIndex()].ID
+				tickets := (*pg.tickets).Confirmed[walletID]
+				return layout.Stack{Alignment: layout.N}.Layout(gtx,
+					layout.Expanded(func(gtx C) D {
+						return layout.Inset{Top: values.MarginPadding60}.Layout(gtx, func(gtx C) D {
+							return c.theme.Card().Layout(gtx, func(gtx C) D {
+								gtx.Constraints.Min = gtx.Constraints.Max
+								if pg.ticketTypeDropDown.SelectedIndex()-1 != -1 {
+									tickets = filterTickets(tickets, func(ticketStatus string) bool {
+										return ticketStatus == strings.ToUpper(pg.ticketTypeDropDown.Selected())
+									})
+								}
+
+								if len(tickets) == 0 {
+									txt := c.theme.Body1("No tickets yet")
+									txt.Color = c.theme.Color.Gray2
+									txt.Alignment = text.Middle
+									return layout.Inset{Top: values.MarginPadding15}.Layout(gtx, func(gtx C) D { return txt.Layout(gtx) })
+								}
+								return layout.UniformInset(values.MarginPadding16).Layout(gtx, func(gtx C) D {
+									return pg.ticketsList.Layout(gtx, len(tickets), func(gtx C, index int) D {
+										return ticketRowActivity(gtx, &c, tickets[index], index)
+									})
+								})
+							})
+						})
+					}),
+					layout.Stacked(func(gtx C) D {
+						return pg.dropDowns(gtx)
+					}),
+				)
+			},
+		}
+		return c.SubPageLayout(gtx, page)
+	}
+
+	return c.Layout(gtx, func(gtx C) D {
+		return c.UniformPadding(gtx, body)
+	})
+}
+
+func (pg *ticketsActivityPage) dropDowns(gtx layout.Context) layout.Dimensions {
+	gtx.Constraints.Min.X = gtx.Constraints.Max.X
+	return layout.Flex{Axis: layout.Horizontal, Spacing: layout.SpaceBetween}.Layout(gtx,
+		layout.Rigid(func(gtx C) D {
+			return pg.walletDropDown.Layout(gtx)
+		}),
+		layout.Rigid(func(gtx C) D {
+			return layout.Flex{Axis: layout.Horizontal}.Layout(gtx,
+				layout.Rigid(func(gtx C) D {
+					return layout.Inset{
+						Left: values.MarginPadding5,
+					}.Layout(gtx, func(gtx C) D {
+						return pg.ticketTypeDropDown.Layout(gtx)
+					})
+				}),
+				layout.Rigid(func(gtx C) D {
+					return layout.Inset{
+						Left: values.MarginPadding5,
+					}.Layout(gtx, func(gtx C) D {
+						return pg.orderDropDown.Layout(gtx)
+					})
+				}),
+			)
+		}),
+	)
+}
+
+func filterTickets(tickets []wallet.Ticket, f func(string) bool) []wallet.Ticket {
+	t := make([]wallet.Ticket, 0)
+	for _, v := range tickets {
+		if f(v.Info.Status) {
+			t = append(t, v)
+		}
+	}
+	return t
+}
+
+func (pg *ticketsActivityPage) handle() {
+	c := pg.common
+	initWalletDropDown(c, &pg.walletDropDown)
+
+	sortSelection := pg.orderDropDown.SelectedIndex()
+	if pg.filterSorter != sortSelection {
```

Comment:
Looks like this filterSorter check is for the purpose of only executing the ticket filter code when the dropdown is clicked. A better implementation is to have a Clicked() event emitted by the dropdown widget when the dropdown widget is clicked. You can look at how the Clicked() method is implemented in the clickable widget.